src/CMakeLists.txt

```diff
@@ -525,34 +547,45 @@ set(zeek_SRCS
     ${FLEX_Scanner_OUTPUTS}
     ${FLEX_Scanner_INPUT}
     ${BISON_Parser_INPUT}
-    ${CMAKE_CURRENT_BINARY_DIR}/DebugCmdConstants.h
     ${CMAKE_CURRENT_BINARY_DIR}/ZAM-MethodDecls.h
     ${THIRD_PARTY_SRCS}
-    ${HH_SRCS}
     ${MAIN_SRCS}
 )
 
 collect_headers(zeek_HEADERS ${zeek_SRCS})
 
-add_library(zeek_objs OBJECT ${zeek_SRCS})
-
-if (ZEEK_STANDALONE)
-    add_executable(zeek main.cc
-                $<TARGET_OBJECTS:zeek_objs>
-                ${zeek_HEADERS}
-                ${bro_SUBDIR_LIBS}
-                ${bro_PLUGIN_LIBS}
+# Add the sources for the core functionality.
+add_library(zeek_src_obj OBJECT ${zeek_SRCS})
+target_link_libraries(zeek_src_obj PRIVATE $<BUILD_INTERFACE:zeek_internal>)
+add_dependencies(zeek_src_obj zeek_autogen_files)
+add_clang_tidy_files(${zeek_SRCS})
+zeek_add_obj_lib(zeek_src_obj)
+
+if (TARGET zeek_exe)
+    target_sources(
+        zeek_exe
+        PRIVATE
+            main.cc
+            ${zeek_HEADERS}
+            ${bro_SUBDIR_LIBS}
+            ${bro_PLUGIN_LIBS})
+    target_link_libraries(
+        zeek_exe
+        PRIVATE
+            ${bro_PLUGIN_LINK_LIBS}
```

Comment:
Maybe something like this https://github.com/robotology/yarp/blob/ea6ed180ddb102679d8b8492fc40329719d7b8e4/cmake/YarpRenamedOption.cmake ?

Reply:
I'll make a pass to see what `bro` variables we can safely get rid off. Some of the variables are used by the dynamic plugins, so changing them would break existing modules.